MonkeyLoader.Resonite.Integration/DataFeeds/Settings/SettingsViewData.cs

```diff
@@ -0,0 +1,353 @@
+﻿using Elements.Assets;
+using Elements.Core;
+using EnumerableToolkit;
+using FrooxEngine;
+using FrooxEngine.UIX;
+using MonkeyLoader.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace MonkeyLoader.Resonite.DataFeeds.Settings
+{
+    /// <summary>
+    /// Stores data about the <see cref="FrooxEngine.SettingsDataFeed"/>
+    /// <see cref="FrooxEngine.RootCategoryView">Views</see> in Userspace - there can be any number of these.
+    /// </summary>
+    public sealed class SettingsViewData
+    {
+        private const string LegacyInjectedColorXTemplateName = "Injected DataFeedValueField<colorX>";
+        private bool _legacyColorXTemplateCleanupDone;
+        private RootCategoryView? _rootCategoryView = null;
+        private Slider<float>? _scrollSlider = null;
+
+        [MemberNotNullWhen(true, nameof(Mapper))]
+        public bool HasMapper => Mapper is not null;
+
+        /// <summary>
+        /// Gets whether this settings view data has a <see cref="RootCategoryView">RootCategoryView</see>.
+        /// </summary>
+        [MemberNotNullWhen(true, nameof(RootCategoryView))]
+        public bool HasRootCategoryView => RootCategoryView is not null;
+
+        public DataFeedItemMapper? Mapper => RootCategoryView?.ItemsManager.TemplateMapper.Target.FilterWorldElement();
+
+        public RootCategoryView? RootCategoryView
+        {
+            get => _rootCategoryView = _rootCategoryView.FilterWorldElement();
+            internal set
+            {
+                if (ReferenceEquals(_rootCategoryView, value))
+                    return;
+
+                CleanupView();
+                _rootCategoryView = value;
+                SetupView();
+            }
+        }
+
+        public SettingsDataFeed? SettingsDataFeed => RootCategoryView?.Feed.Target as SettingsDataFeed;
+
+        [MemberNotNullWhen(true, nameof(ScrollSlider))]
+        internal bool HasScrollSlider => ScrollSlider is not null;
+
+        internal Stack<float> ScrollAmounts { get; } = new();
+
+        internal Slider<float>? ScrollSlider
+        {
+            get => _scrollSlider = _scrollSlider.FilterWorldElement();
+            set => _scrollSlider = value;
+        }
+
+        private static Logger Logger => MonkeyLoaderRootCategorySettingsItems.Logger;
+
+        internal SettingsViewData(SettingsDataFeed dataFeed)
+        {
+            if (!dataFeed.World.IsUserspace())
+                return;
+
+            RootCategoryView = dataFeed.Slot.GetComponent<RootCategoryView>();
+
+            dataFeed.RunSynchronouslyAsync(async () =>
+            {
+                foreach (var primitiveType in GetTemplateTypes())
+                {
+                    await default(NextUpdate);
+                    EnsureDataFeedValueFieldTemplate(primitiveType);
```

Comment:
This causes a significant amount of slowdown on startup despite the template generation being spaced out. I think it was better before when the templates were generated on-demand.

Reply:
Yea... We can move it back to be on demand too, just need to call that method when making the value fields as it used to be